sklearn/_build_utils/__init__.py

```diff
@@ -75,3 +75,26 @@ def maybe_cythonize_extensions(top_path, config):
             nthreads=n_jobs,
             compile_time_env={'SKLEARN_OPENMP_SUPPORTED': with_openmp},
             compiler_directives={'language_level': 3})
+
+
+def gen_from_templates(templates):
+    """Generate cython files from templates"""
+    if not isinstance(templates, list):
+        templates = [templates]
+
+    for template in templates:
+        pyxfile = template.replace('.tp', '')
```

Comment:
super nit but the helper may also generate pxd files, not just pyx

Reply:
I does. You can see it in action in the diff of sklearn.utils.setup.py
I rename `pyxfile` to `outfile` to avoid the confusion.